go/consensus/tendermint/apps/staking/state/state.go

```diff
@@ -717,27 +719,90 @@ func (s *MutableState) TransferFromCommon(
 	if err != nil {
 		return false, fmt.Errorf("tendermint/staking: failed to transfer from common pool: %w", err)
 	}
+	if transferred.IsZero() {
+		// Common pool has been depleated, nothing to transfer.
+		return false, nil
+	}
+
+	// If escrow is requested, escrow the transferred stake immediately.
+	if escrow {
```

Comment:
Hm I think (not 100% sure, didn't check in detail) we can now also simplify the `AddRewards` method by calling `TransferFromCommon` from the accounts loop.